### ISSUE TITLE
PlotSignal maxRenderIndex drawing issue in last unit

### ISSUE THREAD

**Issue by at2software** [ScottPlot/ScottPlot#744]
There seems to be a drawing issue that happens with a `PlotSignal` using `maxRenderIndex`. On our end it can be reproduced when the there is less than one unit of space available on the right side. Example: 50 samples, maxRenderIndex set to 35. Now, if there is space on the X axis up to 36 or more, everything is fine; however if the last fitting value on the X axis is 35 itself, suddenly one additional sample is drawn on the graph. Showing a video probably explains it better:

![maxrenderindex](https://user-images.githubusercontent.com/8616861/106899861-25f91180-66f6-11eb-9b47-ede8ad761e60.gif)

The following code was used for above screen cap:
```cs
private void Window_Loaded(object sender, RoutedEventArgs e)
{
	double[] data = new double[100];

	Random rand = new Random(0);
	for (int i = 1; i <= 50; i++)
		data[i] = data[i - 1] + rand.NextDouble() - .5;

	plot.plt.PlotSignal(data, maxRenderIndex: 35);
	plot.Render();
}
```

The issue can also be observed in the "growing data" example of the current demo, both WinForms and WPF. Here, on the first few samples, there is one additional sample drawn as well, which is a 0 value:

![demo](https://user-images.githubusercontent.com/8616861/106899962-432de000-66f6-11eb-8bbf-c8f0b6dafc74.gif)

I think in the growing data demo, the issue only occurs in the first few samples as at some point, the horizontal margin is larger than one X unit.

Tested in versions 4.0.48 and 4.1.5-beta.

**Comment by StendProg** [comment]
Hi @at2software,

First of all, thanks for the detailed report. This helped a lot in identifying the problem.

PR #745 fix that bug.

**Comment by at2software** [comment]
Thanks for the fast fix!

**Comment by swharden** [comment]
Thanks for reporting this @at2software! And @StendProg, thank you for the quick fix as well.

I applied the same fix to ScottPlot 4.0 (275625d193335b1cecd5e3631a283ce1489cffcc) and 4.1 (8e468c4d949fe3d5473d5c9e0204c7e0fbe63c75). I'll release new NuGet packages for both some time this weekend.
